app/services/katello/ui_notifications/pulp/proxy_disk_space.rb

```diff
@@ -0,0 +1,39 @@
+module Katello
+  module UINotifications
+    module Pulp
+      class ProxyDiskSpace
+        class << self
+          def deliver!
+            SmartProxy.unscoped.with_features('Pulp').each do |proxy|
```

Comment:
You probably want to check Smart Proxies with Pulp and PulpNode feature (PulpNode feature is used for external proxies).  You can use `SmartProxy.with_content` to query both